### PR TITLE
[Refactor] Refactor hive metadata

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/11349

## Problem Summary(Required) ：

Major changes:

* use catalog name instead of resource name. Remove the trick that using `hive.metastore.uris` instead of resource name.
* use HiveMetadataFactory to create query-level hive metadata.

I retain the original logic of HiveMetadata, and I will delete it when the next PR is adapted.
- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 103 (95.15%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|36|39|92.31%|[92, 113, 114]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnectorInternalMgr.java|27|28|96.43%|[99]|
|:large_blue_circle:|com/starrocks/external/hive/CachingHiveMetastoreConf.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/CachingRemoteFileConf.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnector.java [hunk 1]
```diff
@@ -3,58 +3,61 @@
...
+                metastore,
+                remoteFileIO,
+                internalMgr.getHiveMetastoreConf(),
+                internalMgr.getRemoteFileConf(),
+                internalMgr.getPullRemoteFileExecutor(),
+                internalMgr.isSearchRecursive()
+        );
+    }
 
     public void onCreate() {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnector.java:56]: why not remove this function?

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnector.java:56]: After we have adapted auto sync metadata, we still need this operation. Daily test doesn't include auto sync's case. so I just temporarily delete this part of code

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 2]
```diff
@@ -65,12 +75,12 @@ default Table getTable(String dbName, String tblName) {
...
-     * @param databaseName the name of the database
-     * @param tableName the name of the table
-     * @return a list of partition names
+     * Get the remote file information from hdfs or s3. It is mainly used to generate ScanRange for scheduling.
+     * @param table
+     * @param partitionKeys selected columns
+     * @return the remote file information of the query to scan.
      */
-    default List<String> getPartitionNames(String databaseName, String tableName) {
+    default List<RemoteFileInfo> getRemoteFileInfos(Table table, List<PartitionKey> partitionKeys) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:83]: why do we need this API

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 3]
```diff
@@ -53,6 +53,16 @@ default List<String> listTableNames(String dbName) throws DdlException {
         return Lists.newArrayList();
     }
 
+    /**
+     * Return all partition names of the table.
+     * @param databaseName the name of the database
+     * @param tableName the name of the table
+     * @return a list of partition names
+     */
+    default List<String> listPartitionNames(String databaseName, String tableName) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:62]: why do we need this API

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:83]: this needed in building scanRange or HiveTable.tothrift

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 3 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:62]: this is required in prune partition by MetadataMgr.listPartitionNames

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 103 (95.15%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|36|39|92.31%|[92, 113, 114]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnectorInternalMgr.java|27|28|96.43%|[99]|
|:large_blue_circle:|com/starrocks/external/hive/CachingHiveMetastoreConf.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/CachingRemoteFileConf.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:83]: This api isn't very common. I think we will refactor it later. Currently only hive and hudi are needed, I think it is enough at present.

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 102 / 107 (95.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|36|39|92.31%|[92, 113, 114]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnectorInternalMgr.java|31|32|96.88%|[98]|
|:large_blue_circle:|com/starrocks/external/hive/CachingHiveMetastoreConf.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/CachingRemoteFileConf.java|2|2|100.00%|[]|

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Review by Youngwb — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 109 / 124 (87.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|42|55|76.36%|[67, 68, 80, 81, 99, 100, 101, 116, 117, 118, 133, 154, 155]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnectorInternalMgr.java|31|32|96.88%|[98]|
|:large_blue_circle:|com/starrocks/external/hive/CachingHiveMetastoreConf.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/CachingRemoteFileConf.java|2|2|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**